Automattic/wp-calypso
# Reader Lists: Add item adder component

#### Changes proposed in this Pull Request

* Adds a new ItemAdder component for adding new sites to a list. This component consists of three pieces:
  1. A text input for receiving the user's query.
  2. A add/remove toggle button which appears if the query is recognized as a URL.
  3. Search results rendered as cards; each card can be added to the list via a CTA button.

#### Testing instructions

1. Navigate to a Reader List of your choice and navigate to the Sites tab. 
<img width="741" alt="Screen Shot 2020-11-06 at 3 37 29 PM" src="https://user-images.githubusercontent.com/4044428/98421168-fc244600-2045-11eb-8051-97a5214fc42d.png">
2. Enter a query into the search input. Ensure that search results appear below the input. 
<img width="741" alt="Screen Shot 2020-11-06 at 3 38 11 PM" src="https://user-images.githubusercontent.com/4044428/98421215-152cf700-2046-11eb-9d25-a03630bab9ee.png">
3. Add some of the search result feeds into the reader list. Ensure that they are shown in the Added sites section. 
<img width="735" alt="Screen Shot 2020-11-06 at 3 43 56 PM" src="https://user-images.githubusercontent.com/4044428/98421580-e3686000-2046-11eb-9f67-165159507b99.png">
4. Remove sites from the Added sites section. Ensure that a confirmation dialogue is shown before processing the removal. Ensure that the site is removed from the list by dismissing the dialogue with the Remove button. 
<img width="496" alt="Screen Shot 2020-11-06 at 3 40 25 PM" src="https://user-images.githubusercontent.com/4044428/98421349-663ceb00-2046-11eb-8f74-4f26198f7747.png">
5. Enter an RSS feed address to the search input. Ensure that a button appears below the input for following the feed. 
<img width="735" alt="Screen Shot 2020-11-06 at 3 42 17 PM" src="https://user-images.githubusercontent.com/4044428/98421489-a8662c80-2046-11eb-98ad-1c119a4fe892.png">
6. Click on the button and ensure that the feed gets added to the list. Clicking the button again will remove the feed from the list. 
<img width="735" alt="Screen Shot 2020-11-06 at 3 43 03 PM" src="https://user-images.githubusercontent.com/4044428/98421531-c338a100-2046-11eb-8a07-ca264b04692f.png">
